_posts/2021-04-21-frequency-band-control-in-unetstack.md

```diff
@@ -0,0 +1,199 @@
+
+## Understanding baseband signal generation, transmission 
+
+It's good to have a basic understanding of how baseband signals are created and their frequency content. We can try out how it works in Subnero Modems using some examples.
+
+We shall use Binary Phase Shift Keying (BPSK) for this exercise.
+
+Create a baseband complex signal as follows. A complex baseband signal point [1,0] refers to the real and imaginary components of one baseband sample. So in the following signal, all imaginary values are 0.
+```
+> s = [1,0]*12000
+[1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1 <<snip>> 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0]
+
+```
+
+Transmit this 1s long signal (12000 samples at a sample rate of 12kHz).
+
+```
+> phy << new TxBasebandSignalReq(signal: s)
+AGREE
+phy >> TxFrameStartNtf:INFORM[txTime:100610682 txDuration:1000416]
+phy >> TxFrameNtf:INFORM[txTime:100593099]
+```
+You should hear a tone sent out by the Unet audio running in the laptop.
+
+Passband PSD plot from the diagnostics scope shows this carrier only signal at nearly 12kHz.
+
+(settings used: plvl=-25, plot trigger set to -60, you can experiment with power and trigger to get a snapshot depending on your environment)
+
+<p align="center"><img src="../assets/img/freqBandControl/tone.png" style="zoom:50%;"/></p>
+
+The following will send out a 1s BPSK signal with constellation (-1, 1) at the highest rate possible, i.e., every alternating complex sample is (+1,0) followed by (-1,0). 
+
+```
+> s = [1,0, -1, 0]*6000
+[1, 0, -1, 0, 1, 0, -1, 0, 1, 0, <<snip>> -1, 0, 1, 0, -1, 0, 1, 0, -1, 0]
+
+> phy << new TxBasebandSignalReq(signal: s)
+
+```
+
+Passband PSD plot from the diagnostics scope showing the main frequencies at 6kHz and 18kHz implying full use of the band 6 to 18kHz.
+<p align="center"><img src="../assets/img/freqBandControl/bpsk-high.png" style="zoom:50%;"/></p>
+
+
+The following will send out a 1s BPSK signal at a half the highest rate possible. 
+
+```
+> s = [1,0, 1, 0, -1, 0, -1, 0]*3000
+[1, 0, -1, 0, 1, 0, -1, 0, 1, 0, <<snip>> -1, 0, 1, 0, -1, 0, 1, 0, -1, 0]
+
+> phy << new TxBasebandSignalReq(signal: s)
+
+```
+
+Passband PSD plot from the diagnostics scope showing the main frequencies at 9kHz and 15kHz implying use of 6kHz bandwidth
+<p align="center"><img src="../assets/img/freqBandControl/bpsk-low.png" style="zoom:50%;"/></p>
+
+Note that in the strict sense, unless a band limiting RRC (Raised Root Cosine) filter or equivalent is used, the effective bandwidth does not reduce correctly for BPSK with random binary data. The proper implementation of BPSK or general PSK is not aimed in this blog.
+
+The above is intended to give the reader a flavour of how frequency band used changes with signals. The reader no doubt will be interested in testing different types of modulation schemes, and using the PSD scope, the frequency band used can be visualized. Surely it can also be estimated from communication theory.
+
+## OFDM frequency band control
```

Comment:
Want to mention it is typically configured for use as the default DATA channel in UnetStack modems?